src/game.js

```diff
@@ -0,0 +1,112 @@
+    this.shuffle = this.shuffle.bind(this);
+    this.moveToEmpty = this.moveToEmpty.bind(this);
+    this.checkTheEndOfGame = this.checkTheEndOfGame.bind(this);
+  }
+
+  init() {
+    const gameContainer = document.querySelector('.game-begin');
+    const startGame = document.createElement('BUTTON');
+    startGame.innerHTML = 'Start new Game';
+    startGame.classList.add('start-game');
+
+    this.container.classList.add('container');
+    gameContainer.append(startGame, this.container);
+
+    if (JSON.parse(localStorage.getItem('tiles'))) {
+      this.tiles = JSON.parse(localStorage.getItem('tiles'));
+      this.renderTiles(this.tiles);
+    } else {
+      this.renderTiles(this.tilesWin);
+    }
+
+    startGame.addEventListener('click', this.beginGame);
+  }
+
+  beginGame() {
+    this.tiles = [...this.tilesWin];
+    this.shuffle();
+    this.renderTiles(this.tiles);
+  }
+
+  shuffle() {
+    for (let i = this.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      const temp = this.tiles[j];
+      this.tiles[j] = this.tiles[i];
+      this.tiles[i] = temp;
+    }
+    return this.tiles;
+  }
+
+  renderTiles(arr) {
+    this.container.innerHTML = '';
+    arr.forEach((tile) => {
+      const tileCard = new Tile(tile);
+      tileCard.render(this.container);
+    });
+
+    this.container.addEventListener('click', this.moveToEmpty);
+    localStorage.setItem('tiles', JSON.stringify(this.tiles));
+  }
+
+  async moveToEmpty(e) {
+    const moveKey = e.target.closest('DIV');
+
+    const tileToMove = this.tiles.findIndex((item) => item === Number(moveKey.dataset.key));
+
+    if ((this.tiles[tileToMove - 1] === '') && (tileToMove % 4 !== 0)) {
+      const bingo = this.tiles[tileToMove - 1];
+      this.tiles[tileToMove - 1] = this.tiles[tileToMove];
+      this.tiles[tileToMove] = bingo;
+    }
+    if ((this.tiles[tileToMove + 1] === '') && ((tileToMove + 1) % 4 !== 0)) {
+      const bingo = this.tiles[tileToMove + 1];
+      this.tiles[tileToMove + 1] = this.tiles[tileToMove];
+      this.tiles[tileToMove] = bingo;
+    }
+    if (this.tiles[tileToMove - 4] === '') {
+      const bingo = this.tiles[tileToMove - 4];
+      this.tiles[tileToMove - 4] = this.tiles[tileToMove];
+      this.tiles[tileToMove] = bingo;
+    }
+    if (this.tiles[tileToMove + 4] === '') {
+      const bingo = this.tiles[tileToMove + 4];
+      this.tiles[tileToMove + 4] = this.tiles[tileToMove];
+      this.tiles[tileToMove] = bingo;
+    }
+
+    try {
+      await localStorage.setItem('tiles', JSON.stringify(this.tiles));
+      await this.renderTiles(this.tiles);
+      await this.checkTheEndOfGame();
+    } catch (err) {
+      console.log(new Error('Sorry for inconvenience. Something went wrong'));
+    }
+  }
+
+  checkTheEndOfGame() {
+    if ((JSON.stringify(this.tiles) === JSON.stringify(this.tilesWin))
+        || (JSON.stringify(this.tiles) === JSON.stringify(this.tilesWin1514))) {
```

Comment:
What is `this.tilesWin1514` ??

Reply:
I think we need to add new property in constuructor to check if user is winner or not, and if yes, change this property here